### PR TITLE
Allow custom atols using scoped values

### PR DESCRIPTION
**Pull request by JoshuaLampert**
As discussed on Zulip. To set a preference you need to run, e.g., 
```julia
julia> using Preferences

julia> using Meshes

julia> set_preferences!(Meshes, "atol_float32" => 1.0f-6)
```
and restart the REPL. This writes the preferences to a file "LocalPreferences.toml".

### CODE REVIEW

**Review by juliohm — COMMENTED** [review]
That is really nice. Appreciate if you can add a section to the docs explaining this workflow.

**Comment by JoshuaLampert** [comment]
I added a section in the docs.

**Comment by JoshuaLampert** [comment]
We decided to opt for ScopedValues.jl instead of Preferences.jl. I updated the PR accordingly.

File: docs/make.jl [hunk 1]
```diff
@@ -34,7 +34,8 @@ makedocs(
       "Transforms" => "transforms.md",
       "Random" => "rand.md",
       "Visualization" => "visualization.md",
-      "Input/Output" => "io.md"
+      "Input/Output" => "io.md",
+      "Changing default tolerances" => "tolerances.md"
```
Comment (juliohm) [inline docs/make.jl]:
```suggestion
      "Tolerances" => "tolerances.md"
```

File: docs/src/tolerances.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Changing default absolute tolerances
```
Comment (juliohm) [inline docs/src/tolerances.md]:
```suggestion
# Tolerances
```

**Review by juliohm — APPROVED** [review]
Looks great!

File: src/tolerances.jl [hunk 3]
```diff
@@ -15,8 +18,8 @@ isapprox(a::T, b::T, atol=atol(T))
 ```
 """
 atol(x) = atol(typeof(x))
-atol(::Type{Float64}) = 1.0e-10
-atol(::Type{Float32}) = 1.0f-5
+atol(::Type{Float64}) = ATOL64[]
+atol(::Type{Float32}) = ATOL32[]
```
Comment (DanielVandH) [inline src/tolerances.jl:22]:
```suggestion
@inline atol(::Type{Float64}) = ATOL64[]::Float64
@inline atol(::Type{Float32}) = ATOL32[]::Float32
```

File: src/tolerances.jl [hunk 4]
```diff
@@ -2,6 +2,9 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+ATOL64 = ScopedValue(1e-10)
+ATOL32 = ScopedValue(1f-5)
```
Comment (DanielVandH) [inline src/tolerances.jl]:
```suggestion
const ATOL64 = ScopedValue(1e-10)
const ATOL32 = ScopedValue(1f-5)
```

**Review by juliohm — APPROVED** [review]

**Pull request merged**